ðŸ‘‹ Hi, Iâ€™m @lhallee

My name is [Logan Hallee](https://www.linkedin.com/in/logan-hallee/), a PhD Candidate in Bioinformatics Data Science at the University of Delaware ([Gleghorn Lab](https://www.gleghornlab.com/)) specializing in curating large-dimensional feature spaces for biological data. My primary focus is on protein design and annotation using transformer neural networks. Techniques I developed led to the creation of [SYNTERACT](https://huggingface.co/GleghornLab/SYNTERACT), the first large language model approach to protein-protein interaction prediction, ranking in the [top 3% of research outputs](https://www.biorxiv.org/content/10.1101/2023.06.07.544109v1.article-metrics) by Altmetric.

At the Wolfram Winter School, I collaborated with Stephen Wolfram and other mentors to create "[Tetris For Proteins](https://community.wolfram.com/groups/-/m/t/2777595
)," a shape-based metric for protein-protein interactions that emulates lock-and-key enzyme-substrate dynamics, generating hypotheses about protein aggregation likelihood.

I created the [Annotation Vocabulary](https://www.biorxiv.org/content/10.1101/2024.07.30.605924v1.abstract), a unique set of integers mapped to popular protein and gene ontologies, enabling state-of-the-art protein annotation and generation models when used with its own token embedding.

My work also supports the paradigm of codon usage bias as a key biological phenomenon for phylogenetic analysis. Our models, published in [Nature Scientific Reports](https://www.nature.com/articles/s41598-023-28965-7), highlight codon usage as a unique phylogenetic predictor. Our lab recently produced [cdsBERT](https://huggingface.co/GleghornLab/cdsBERT), showcasing cost-effective techniques to enhance the biological relevance of protein language models using a codon vocabulary.

In natural language processing, I invented [Mixture of Experts extension](https://arxiv.org/abs/2401.15713) for scalable transformer networks adept at sentence similarity tasks. We believe future networks with N experts will perform like N independently trained networks, offering significant time and computational savings for vector retrieval systems and search relying on semantic vector representations.

I also manage lab projects in computer vision, utilizing deep learning to reconstruct anatomically accurate 3D organs from 2D Z-stacks, informing morphometric and pharmacokinetic studies.

Some other stuff I've worked on over the years:
- [featureranker](https://github.com/lhallee/featureranker), a Python package for feature ranking
- My textbook section about [Protein Language Models](https://www.researchgate.net/profile/Logan-Hallee/publication/378769504_413_Predicting_Structure_and_Function_of_Biomolecules_Through_Natural_Language_Processing_Tools/links/65e8d39dadc608480a056202/413-Predicting-Structure-and-Function-of-Biomolecules-Through-Natural-Language-Processing-Tools.pdf)
- Machine learning to identify [cardioprotective molecules](https://www.ahajournals.org/doi/abs/10.1161/circ.149.suppl_1.P109) in minority groups
- Writing about the relationships of [Hsp90](https://www.mdpi.com/2075-4426/11/12/1373) and [Gamma secretase](https://www.mdpi.com/2075-4426/11/12/1294) in cardiac diseases

Norway, ME âž” Newark, DE

<!---
lhallee/lhallee is a âœ¨ special âœ¨ repository because its `README.md` (this file) appears on your GitHub profile.
You can click the Preview link to take a look at your changes.
--->
